crates/wasi-common/src/sys/unix/linux/hostcalls_impl.rs

```diff
@@ -98,7 +97,7 @@ pub(crate) fn fd_readdir_impl(
         dir.seek(loc);
     }
 
-    Ok(dir.into_iter().map(|entry| {
+    Ok(DirIter(dir).map(|entry| {
```

Comment:
`into_iter` seems more idiomatic to me, is there any particular reason why you switched to literal `DirIter`?

Reply:
Since it's an internal structure, I wanted to keep `DirIter` private to the module. By implementing an `IntoIterator` for `Dir` we have to leak the struct outside of the module we seems like a lot of noise for little to be gained.

Reply:
Fair enough!